lib/adm.js

```diff
@@ -4127,16 +4127,189 @@ maAdm.prototype.generatePlan = function (opts, callback)
 	setTimeout(callback, 0);
 };
 
+/*
+ * Retrieves Image objects, specified by the uuid keys of the
+ * 'services_by_image' parameter, setting the results in opts.local_images.
+ */
+function fetchLocalImages(opts, cb) {
+	assertplus.object(opts, 'opts');
+	assertplus.object(opts.services_by_image, 'opts.services_by_image');
+	assertplus.object(opts.log, 'opts.log');
+	assertplus.object(opts.local_imgapi, 'opts.local_imgapi');
+	assertplus.func(cb, 'cb');
+
+	opts.local_images = {};
+
+	var imgapi_opts = {};
+	vasync.forEachParallel({
+	    inputs: Object.keys(opts.services_by_image),
+	    func: function getOneImageName(imageUuid, nextImage) {
+		opts.log.debug(
+			    'getting local image %s for service %s',
+			    imageUuid, opts.services_by_image[imageUuid]);
+		opts.local_imgapi.getImage(imageUuid, imgapi_opts,
+			function getImage(err, image) {
+			if (err) {
```

Comment:
This should probably differentiate from a 404 (ResourceNotFound) an some other response error. Here is comparable code in sdcadm that does that:

```
                    self.imgapi.getImage(imgUuid, function (err, img) {
                        if (!err) {
                            ctx.imgFromUuid[imgUuid] = img;
                            nextImg();
                        } else if (err.restCode !== 'ResourceNotFound') {
                            nextImg(new errors.SDCClientError(
                                err, 'imgapi'));
                        } else {
                            nextImg();
                        }
                    });
```

Reply:
Yep, will do.

Reply:
Thanks, added that in the latest commit.